config/config.go

```diff
@@ -36,55 +49,88 @@ type Config struct {
 		Format string
 		Level  string
 	}
-
-	Core struct {
-		Image string
-	}
 }
 
 // New creates a new config with default values.
-func New() *Config {
+func New() (*Config, error) {
+	home, err := homedir.Dir()
+	if err != nil {
+		return nil, err
+	}
+
 	var c Config
 	c.Server.Address = ":50052"
 	c.Client.Address = "localhost:50052"
+	c.Database.Path = filepath.Join(home, ".mesg", "db")
 	c.Log.Format = "text"
 	c.Log.Level = "info"
-	c.Core.Image = "mesg/core:" + strings.Split(version.Version, " ")[0]
-	return &c
+	return &c, nil
 }
 
 // Global returns a singleton of a Config after loaded ENV and validate the values.
 func Global() (*Config, error) {
+	var err error
 	once.Do(func() {
-		instance = New()
-		instance.Load()
+		instance, err = New()
+		if err != nil {
+			return
+		}
+		instance.LoadFromEnv()
 	})
+	if err != nil {
+		return nil, fmt.Errorf("config: %s", err)
+	}
+
 	if err := instance.Validate(); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("config: %s", err)
 	}
 	return instance, nil
 }
 
-// Load reads config from environmental variables.
-func (c *Config) Load() {
+// Load reads config from yaml and env.
+// Note that env variables have higher precedence then yaml config.
+func (c *Config) Load(file string) error {
+	if err := c.LoadFromYaml(file); err != nil {
```

Comment:
Didn't we say that for now we will just keep it simple with the support of env variables only and we can always add a `.envrc` if we want to have a "config file". Moreover I don't see where this function is used.

Reply:
> Didn't we say that for now we will just keep it simple with the support of env variables only and we can always add a .envrc if we want to have a "config file".

Yes, we did. We said also that in the future we will have yaml config. I have added yaml parser to support Plugin slice - passing such complicated struct in env is less readable. Envconf support slices but I guess it's only for basic go types and not for the structs.

> Moreover I don't see where this function is used.

That's true. It's not used yet - it will be if we want to.

Reply:
I agree with Antho, let's keep it simple for now. So as we don't load any config file, why keeping the function.